dashboard/app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Api::V1::UsersController < ApplicationController
+  before_action :load_user
+
+  def load_user
+    user_id = params[:user_id]
+    @user = current_user
+    if user_id != 'me' && user_id != current_user.id
+      raise CanCan::AccessDenied
+    end
+  end
+
+  # GET /api/v1/users/<user_id>/using_text_mode
+  def get_using_text_mode
+    render json: {using_text_mode: @user.using_text_mode}
+  end
+
+  # POST /api/v1/users/<user_id>/using_text_mode
+  def post_using_text_mode
+    @user.using_text_mode = params[:using_text_mode].to_bool
```

Comment:
minor: this will blow up with `NoMethodError: undefined method `to_bool' for nil:NilClass` if the param is missing. You could do something like `!!params[:using_text_mode].try(:to_bool)`

Note `nil.try(:to_bool)` returns nil, hence the `!!`. If nil is acceptable (interpreted as false), then you don't need that.